tests/dhcp_relay/test_dhcp_pkt_recv.py

```diff
@@ -36,17 +37,27 @@ def check_dhcp_relay_feature_state(duthost):
 class Dhcpv6PktRecvBase:
 
     @pytest.fixture(scope="class")
-    def setup_teardown(self, duthost, tbinfo):
+    def setup_teardown(self, rand_selected_dut, tbinfo):
         disabled_host_interfaces = tbinfo['topo']['properties']['topology'].get('disabled_host_interfaces', [])
         ptf_indices = [interface for interface in tbinfo['topo']['properties']['topology'].get('host_interfaces', [])
                        if interface not in disabled_host_interfaces]
-        dut_intf_ptf_index = duthost.get_extended_minigraph_facts(tbinfo)['minigraph_ptf_indices']
+        dut_intf_ptf_index = rand_selected_dut.get_extended_minigraph_facts(tbinfo)['minigraph_ptf_indices']
+        if 'dualtor' in tbinfo['topo']['name']:
+            pattern = r'0.(\d+)'
```

Comment:
I don't think this covers all the topologies. Ex: `0.0@0,1.0@0` (dualtor-120 topology)

Also hardcoding `0.` means we are considers first duthost (uppper tor), so theoretically this is incorrect (we need to get the ptf_indices for the rand_selected_dut).